src/AsyncMqttClient.cpp

```diff
@@ -335,20 +335,19 @@ void AsyncMqttClient::_onDisconnect(AsyncClient* client) {
   (void)client;
   AsyncMqttClientDisconnectReason reason;
 
-  if (_connectPacketNotEnoughSpace) {
-    reason = AsyncMqttClientDisconnectReason::ESP8266_NOT_ENOUGH_SPACE;
-  } else if (_tlsBadFingerprint) {
-    reason = AsyncMqttClientDisconnectReason::TLS_BAD_FINGERPRINT;
-  } else {
-    reason = AsyncMqttClientDisconnectReason::TCP_DISCONNECTED;
-  }
+  if (!_disconnectFlagged) {
+    AsyncMqttClientDisconnectReason reason;
```

Comment:
Isn't it confusing defining local reason variable here with the same name as on previous scope?

Reply:
Ooops my bad, was working on a new feature when found this problem, and my cherry-picking glitched... :P

Let me fix that